eng/common/scripts/Validate-All-Packages.ps1

```diff
@@ -1,50 +1,252 @@
-        exit 1
+    Write-Host "Checking if a version is already shipped for package $packageName with version $packageVersion."
+    $parsedNewVersion = [AzureEngSemanticVersion]::new($packageVersion)
+    $versionMajorMinor = "" + $parsedNewVersion.Major + "." + $parsedNewVersion.Minor
+    $workItem = FindPackageWorkItem -lang $LanguageDisplayName -packageName $packageName -version $versionMajorMinor -includeClosed $true -outputCommand $false
+    if ($workItem)
+    {
+        # Check if the package version is already shipped
+        $shippedVersionSet = ParseVersionSetFromMDField $workItem.fields["Custom.ShippedPackages"]
+        if ($shippedVersionSet.ContainsKey($packageVersion)) {
+            return $true
+        }
+    }
+    else {
+        Write-Host "No work item found for package [$packageName]. Creating new work item for package."
     }
+    return $false
 }
 
-# Check if package config file is present. This file has package version, SDK type etc info.
-if (-not $ConfigFileDir) {
-    $ConfigFileDir = Join-Path -Path $ArtifactPath "PackageInfo"
+# Function to create or update package work item in Azure DevOps
+function CreateUpdatePackageWorkItem($pkgInfo)
+{
+    $versionString = $pkgInfo.Version
+    $packageName = $pkgInfo.Name
+    $plannedDate = $pkgInfo.ReleaseStatus
+    $setReleaseState = $true
+    if (!$plannedDate -or $plannedDate -eq "Unreleased")
+    {
+        $setReleaseState = $false
+        $plannedDate = "unknown"
+    }
+
+    # Create or update package work item
+    $succeeded = Update-DevOpsReleaseWorkItem `
+                    -language $LanguageDisplayName `
+                    -packageName $packageName `
+                    -version $versionString `
+                    -plannedDate $plannedDate `
+                    -packageRepoPath $pkgInfo.serviceDirectory `
+                    -packageType $pkgInfo.SDKType `
+                    -packageNewLibrary $pkgInfo.IsNewSDK `
+                    -serviceName "unknown" `
+                    -packageDisplayName "unknown" `
+                    -inRelease $IsReleaseBuild
+
+    if (-not $succeeded) {
+        Write-Host "Update of the Devops Release WorkItem failed."
+    }
+    return $succeeded
 }
-foreach ($artifact in $ArtifactList)
+
+CheckAzLoginAndDevOpsExtensionInstall
+
+CheckDevOpsAccess
+
+Write-Host "Artifact path: $($ArtifactPath)"
+
+foreach ($packageInfoFile in $PackageInfoFiles)
 {
-    Write-Host "Processing $($artifact.name)"
-    ProcessPackage -PackageName $artifact.name -ConfigFileDir $ConfigFileDir
-}
\ No newline at end of file
+    $packageInfo = Get-Content $packageInfoFile | ConvertFrom-Json
+
+    Write-Host "Processing packageInfoFile: $packageInfoFile"
+    # The ArtifactName maps to the PackageInfo
+    Write-Host "Processing $($packageInfo.Name)"
+    Write-Host "Is Release Build: $IsReleaseBuild"
+
+# JRS everything below needs to be updated to pull information from the PackageInfo which was
```

Comment:
I assume you did this based on what I'm seeing.

Reply:
Removed locally, will be gone with the next commit.